### PR TITLE
Write a compiler from Scrapscript to C

### PR DESCRIPTION
**Pull request by tekknolagi**
Emit straight C, no IR. Includes a GC and small runtime.

Has immediate small integers (no bignum), small strings and heap strings, and immediate empty list.

### CODE REVIEW

**Comment by tekknolagi** [comment]
TODO... test it...

File: README.md [hunk 1]
```diff
@@ -53,6 +53,30 @@ docker run -i -t ghcr.io/tekknolagi/scrapscript:trunk apply "1 + 2"
 docker run -i -t ghcr.io/tekknolagi/scrapscript:trunk repl
 ```
 
+### The experimental compiler:
```
Comment (gregorybchris) [inline README.md:56]: This is so awesome!! What do you think about adding some prelim benchmark results? The first thing I'm wondering is whether it's worth using over the interpreter.

Would understand if you want to hold off until you can benchmark something big or standard.

**Review by gregorybchris — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (tekknolagi) [reply README.md:56]: Don't have support for variants yet but would like to add them. Shouldn't be too hard. That unlocks a lot of programs

Comment (tekknolagi) [reply README.md:56]: Then we can benchmark hull.scrap or similar

**Comment by tekknolagi** [comment]
rebased on top of variants-in-trunk

**Pull request merged**